File: robot_fleet/packages/client_sdk/typescript/src/realtime/events.ts
import type { Task } from '../models/types'

export type GatewayRealtimeMessage =
  | { type: 'connected'; message: string }
  | { type: 'invalidate'; queries: string[]; timestamp: number }
  | { type: 'tasks_update'; plan_id: number; tasks: Task[] }
  // Future-proof: unknown messages pass through as generic objects
  | ({ type: string } & Record<string, any>)

File: robot_fleet/packages/client_sdk/python/src/realtime/events.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Literal, Union


@dataclass(frozen=True)
class Connected:
    type: Literal["connected"]
    message: str


@dataclass(frozen=True)
class Invalidate:
    type: Literal["invalidate"]
    queries: list[str]
    timestamp: float


@dataclass(frozen=True)
class TasksUpdate:
    type: Literal["tasks_update"]
    plan_id: int
    tasks: list[dict[str, Any]]


GatewayRealtimeMessage = Union[Connected, Invalidate, TasksUpdate, dict[str, Any]]

